### DISCUSSION TITLE
Suggested file systems and block device options

### DISCUSSION THREAD

**Discussion by mitake** [etcd-io/etcd#15978] [Ideas]
Currently etcd doesn't provide suggested file systems and block device options. Behavior of fs and block device are important e.g. they shouldn't pause for a long time, and it's the motivation of implementing [the watchdog mechanism](https://github.com/etcd-io/etcd/pull/15440). I think it's better to provide suggested options which are aligned with assumptions of etcd. Otherwise possibility of hard to reason behavior might increase, for example:
- default timeout of EBS is [30 seconds](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/nvme-ebs-volumes.html#timeout-nvme-ebs-volumes), it might be too long for typical etcd deployment. Some lease issues can caused by this factor.
- if a user uses ext4 and its has mount option `errors=remount-ro`, file system can be readonly when it sees errors from its block device. In this case, read access can be successful if data is cached by some layer and otherwise causes EIO. It will result hard to reason behavior.

What do you think @ahrtr ?

**Comment by jmhbnz** [comment]
Nice idea @mitake - I think this would be a great addition to our docs.  We have some very basic storage [hardware recommendations](https://etcd.io/docs/v3.5/op-guide/hardware/#disks) we could expand on, and nothing I can find for filesystem side currently.

**Comment by ahrtr** [comment]
Thanks @mitake . It's nice to have, but I am not sure whether it's proper to add such domain (e.g storage in this case) specific info in etcd official guide. The existing [generic guide & requirement on disk](https://etcd.io/docs/v3.5/op-guide/hardware/#disks) might be good enough.

If you have any summary to share, probably you can write a [blog](https://etcd.io/blog/)?

**Comment by mitake** [comment]
Thanks for your feedback @jmhbnz @ahrtr . The existing guide focuses on performance of block devices, and behavior of broken devices is out of scope now. I think we don't have a clear consensus about etcd's assumptions of block device behavior yet, so it's nice if we can 1. figure out the assumptions, then 2. write a doc which explicitly specify. What do you think?
